Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -83,34 +83,56 @@ public class SchemeGenerator {
 
     public func generateScheme(_ scheme: Scheme) throws -> XCScheme {
 
-        func getBuildEntry(_ buildTarget: Scheme.BuildTarget) -> XCScheme.BuildAction.Entry {
+        func getBuildEntry(_ buildTarget: Scheme.BuildTarget) throws -> XCScheme.BuildAction.Entry {
+            let pbxProj: PBXProj
+            let projectFilePath: String
+            switch buildTarget.target.location {
+            case .project(let project):
+                guard let projectReference = self.project.getProjectReference(project) else {
+                    throw SchemeGenerationError.missingProject(project)
+                }
+                pbxProj = try XcodeProj(pathString: projectReference.path).pbxproj
```

Comment:
This could become a performance issue, especially if there are lots of target references. We could at least cache the projects in this class 
```swift
var projects: [String: XcodeProj]
```
As far as I understand you're only loading these projects to get the proper file extension for the target right? Does the buildable name require that? Cause if it doesn't we can just do away with loading the projects.

Reply:
> We could at least cache the projects in this class

Right, I added cache logic.

> As far as I understand you're only loading these projects to get the proper file extension for the target right? Does the buildable name require that? Cause if it doesn't we can just do away with loading the projects.

`PBXProj` is also used to get blueprint of target https://github.com/yonaskolb/XcodeGen/pull/655/files#diff-b0109f56886f826d76bacb6b15dbf86dR132